internal/jwks/manager.go

```diff
@@ -0,0 +1,161 @@
+package jwks
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/rakutentech/jwk-go/jwk"
+	"golang.org/x/sync/singleflight"
+)
+
+const (
+	_defaultRetries = 5
+	_defaultTimeout = 5 * time.Second
```

Comment:
We use 1 second as default value for timeout in proxy handlers, I suggest using the same here